crates/sui-indexer/src/indexer.rs

```diff
@@ -112,8 +112,8 @@ impl Indexer {
         let (exit_sender, exit_receiver) = oneshot::channel();
         // Spawn a task that links the cancellation token to the exit sender
         spawn_monitored_task!(async move {
-            cancel_clone.cancelled().await;
             let _ = exit_sender.send(());
```

Comment:
I'm not sure this order can be swapped. We want to send the `exit_sender` only after `cancel` has been cancelled. Otherwise wouldn't the ingestion executor exit right away?